nordicsemi/dfu/dfu_transport_serial.py

```diff
@@ -342,34 +344,35 @@ def __ensure_bootloader(self):
 
                 trigger.clean()
             if not self.__is_device_in_bootloader_mode(device):
-                logger.info("Serial: Device is either not in bootloader mode, or using an unsupported bootloader.")
+                logger.info(
+                    "Serial: Device is either not in bootloader mode, or using an unsupported bootloader.")
 
     def __is_device_in_bootloader_mode(self, device):
         if not device:
             return False
 
         #  Return true if nrf bootloader or Jlink interface detected.
-        return ((device.vendor_id.lower() == '1915' and device.product_id.lower() == '521f') # nRF52 SDFU USB
-             or (device.vendor_id.lower() == '1366' and device.product_id.lower() == '0105') # JLink CDC UART Port
-             or (device.vendor_id.lower() == '1366' and device.product_id.lower() == '1015'))# JLink CDC UART Port (MSD)
+        return ((device.vendor_id.lower() == '1915' and device.product_id.lower() == '521f')  # nRF52 SDFU USB
+                # JLink CDC UART Port
+                or (device.vendor_id.lower() == '1366' and device.product_id.lower() == '0105')
+                or (device.vendor_id.lower() == '1366' and device.product_id.lower() == '1015'))  # JLink CDC UART Port (MSD)
 
     def __set_prn(self):
         logger.debug("Serial: Set Packet Receipt Notification {}".format(self.prn))
         self.dfu_adapter.send_message([DfuTransportSerial.OP_CODE['SetPRN']]
-            + list(map(ord, struct.pack('<H', self.prn))))
+                                      + list(struct.pack('<H', self.prn)))
         self.__get_response(DfuTransportSerial.OP_CODE['SetPRN'])
 
     def __get_mtu(self):
         self.dfu_adapter.send_message([DfuTransportSerial.OP_CODE['GetSerialMTU']])
         response = self.__get_response(DfuTransportSerial.OP_CODE['GetSerialMTU'])
-
-        self.mtu = struct.unpack('<H', bytearray(response))[0]
+        self.mtu = int(struct.unpack('<H', bytearray(response))[0])
```

Comment:
I need to check again but I think the type of struct.[] was due to a weird reason float and therefore later functions could not deal with it. 

This might be an error on my side since we tried to get the lib working with python3 during a nightly hack session.